### PR TITLE
feat: implement cancel button and modal

### PR DESCRIPTION
**Pull request by katrinan029**
# Description
As part of enhancing the user experience in the provisioning tool for LC2, we need to implement the ability to cancel changes made in the edit form. 

# Solution
The "Cancel" button is added to the top and bottom of the form in edit view. Additionally, a confirmation modal appears if the user leaves the edit mode with unsaved edits, allowing them to confirm their decision before discarding the changes.

https://github.com/openedx/frontend-app-support-tools/assets/71999631/c812dbbe-91ef-40d2-bdfd-4a84c0d981e2

[JIRA](https://2u-internal.atlassian.net/browse/ENT-7506)

### CODE REVIEW

File: src/Configuration/Provisioning/SubsidyEditView/SubsidyEditView.jsx [hunk 1]
```diff
@@ -35,6 +35,19 @@ const SubsidyEditView = () => {
     });
   };
 
+  useEffect(() => {
+    const handleTabClose = (event) => {
+      event.preventDefault();
+      const copyEvent = event;
+      copyEvent.returnValue = true;
+      return copyEvent.returnValue;
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/SubsidyEditView/SubsidyEditView.jsx:46]: Do we need to make a copy of the event here?
```suggestion
      const copyEvent = event;
      copyEvent.returnValue = true;
      return copyEvent.returnValue;
```

Comment (katrinan029) [reply src/Configuration/Provisioning/SubsidyEditView/SubsidyEditView.jsx:46]: Yup! this was to get around the linter rule `no-param-reassign when setting a property on a DOM object`

Comment (marlonkeating) [reply src/Configuration/Provisioning/SubsidyEditView/SubsidyEditView.jsx:46]: Do we need to set this?  Documentation suggests that `returnValue` is true by default (and also deprecated).
https://developer.mozilla.org/en-US/docs/Web/API/Event/returnValue

File: src/Configuration/Provisioning/ProvisioningForm/PolicyForm/ProvisioningFormCatalog.jsx [hunk 2]
```diff
@@ -42,6 +48,9 @@ const ProvisioningFormCatalog = ({ index }) => {
   const handleChange = (e) => {
     const newTabValue = e.target.value;
     const newCatalogQuery = e.target.dataset.catalogqueryid;
+    if (isEditMode && !hasEdits) {
+      setHasEdits(true);
+    }
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/ProvisioningForm/PolicyForm/ProvisioningFormCatalog.jsx:53]: Looks like these lines aren't being covered by tests.

**Review by marlonkeating — CHANGES_REQUESTED** [review]

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**